src/url_aggregator.cpp

```diff
@@ -298,6 +298,13 @@ bool url_aggregator::set_port(const std::string_view input) {
   // Revert changes if parse_port fails.
   uint32_t previous_port = components.port;
   parse_port(trimmed);
```

Comment:
I think this check needs to be done inside parse port. Right now, you're doing the same work twice.